mathesar_ui/src/stores/table-data/columns.ts

```diff
@@ -172,6 +172,11 @@ export class ColumnsDataStore
     return column;
   }
 
+  async rename(id: Column['id'], newName: string): Promise<void> {
+    await this.api.update(id, { name: newName });
+    await this.dispatch('columnRenamed', id);
```

Comment:
Currently when a column is renamed and the rename request succeeds, the values displayed on table for that particular column goes blank until the records request completes.

This would be fixed when the records endpoint uses column id instead of name. I just wanted to make a note of it here.